src/expr.rs

```diff
@@ -418,11 +476,55 @@ impl Expr {
     /// ```
     pub fn matches(self, j: Option<&Value>) -> Result<bool, Error> {
         let reduced = self.reduce(j)?;
+
         match reduced {
             Expr::Bool(v) => Ok(v),
             _ => Err(Error::NonReduced()),
         }
     }
+
+    /// Run CQL against a JSON Value, returning false if the expression does not reduce to a boolean.
+    pub fn matches_or_false(self, j: Option<&Value>) -> Result<bool, Error> {
```

Comment:
This function name is a little confusing to me on first read. I get what it's doing (requiring an exact match) but it feels a bit redundant on `matches`.

Will a user ever want to do anything other than `matches() -> bool`? Do we need to capture those error state variations? I see that it's used in `filter`, but I think `filter` could work with the simple `matches() -> bool` function as well?